File: ts/src/resources/organizations.ts
import { HttpTransport } from "../http.js"

export class OrganizationsResource {
  /**
   * @param base - platform base URL (gateway/platform)
   */
  constructor(private readonly http: HttpTransport, private readonly base: string) {}

  private get v1() { return `${this.base}/v1` }
  private get v2() { return `${this.base}/v2` }

  async list(params?: { limit?: number; skip?: number; is_active?: boolean }) {
    const url = new URL(`${this.v2}/organizations`)
    if (params?.limit)  url.searchParams.set("limit",     String(params.limit))
    if (params?.skip !== undefined) url.searchParams.set("skip", String(params.skip))
    if (params?.is_active !== undefined) url.searchParams.set("is_active", String(params.is_active))
    return this.http.getFetch()(url, { method: "GET" }).then(r => r.json())
  }

  async listAll(params?: { is_active?: boolean }) {
    const url = new URL(`${this.v2}/organizations/_all`)
    if (params?.is_active !== undefined) url.searchParams.set("is_active", String(params.is_active))
    return this.http.getFetch()(url, { method: "GET" }).then(r => r.json())
  }

  async create(body: Record<string, unknown>) {
    return this.http.getFetch()(`${this.v1}/organizations`, {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify(body),
    }).then(r => r.json())
  }
}

File: py/src/imbrace/resources/organizations.py
from typing import Any, Dict
from ..http import HttpTransport, AsyncHttpTransport


class OrganizationsResource:
    """Organizations domain — Sync.

    @param base - platform service base URL (gateway/platform)
    """

    def __init__(self, http: HttpTransport, base: str):
        self._http = http
        self._base = base.rstrip("/")

    @property
    def _v1(self) -> str:
        return f"{self._base}/v1"

    @property
    def _v2(self) -> str:
        return f"{self._base}/v2"

    def list(self, limit: int = 10, skip: int = 0) -> Dict[str, Any]:
        return self._http.request("GET", f"{self._v2}/organizations",
                                  params={"limit": limit, "skip": skip}).json()

    def create(self, body: Dict[str, Any]) -> Dict[str, Any]:
        return self._http.request("POST", f"{self._v1}/organizations", json=body).json()

    def list_all(self) -> Dict[str, Any]:
        return self._http.request("GET", f"{self._v2}/organizations/_all").json()


class AsyncOrganizationsResource:
    """Organizations domain — Async."""

    def __init__(self, http: AsyncHttpTransport, base: str):
        self._http = http
        self._base = base.rstrip("/")

    @property
    def _v1(self) -> str:
        return f"{self._base}/v1"

    @property
    def _v2(self) -> str:
        return f"{self._base}/v2"

    async def list(self, limit: int = 10, skip: int = 0) -> Dict[str, Any]:
        res = await self._http.request("GET", f"{self._v2}/organizations",
                                       params={"limit": limit, "skip": skip})
        return res.json()

    async def create(self, body: Dict[str, Any]) -> Dict[str, Any]:
        res = await self._http.request("POST", f"{self._v1}/organizations", json=body)
        return res.json()

    async def list_all(self) -> Dict[str, Any]:
        res = await self._http.request("GET", f"{self._v2}/organizations/_all")
        return res.json()
